valet/pathproc.go

```diff
@@ -42,35 +43,41 @@ type ProcessParams struct {
 	MaxProc       int           // The maximum number of threads to run
 }
 
-func ProcessFiles(cancelCtx context.Context, params ProcessParams) {
+func ProcessFiles(cancelCtx context.Context, params ProcessParams) error {
 
 	wpaths, werrs := WatchFiles(cancelCtx, params.Root, params.MatchFunc,
 		params.PruneFunc)
 	fpaths, ferrs := FindFilesInterval(cancelCtx, params.Root, params.MatchFunc,
 		params.PruneFunc, params.SweepInterval)
 
 	mpaths := MergeFileChannels(wpaths, fpaths)
-	errs := MergeErrorChannels(werrs, ferrs)
+	merrs := MergeErrorChannels(werrs, ferrs)
 	done := make(chan token)
 
 	log := logs.GetLogger()
+	var err error
 
 	go func() {
 		defer func() { done <- token{} }()
 
-		err := DoProcessFiles(mpaths, params.Plan, params.MaxProc)
-		if err != nil {
-			log.Error().Err(err).Msg("failed processing")
-			os.Exit(1)
-		}
+		err = DoProcessFiles(mpaths, params.Plan, params.MaxProc)
 	}()
 
-	if err := <-errs; err != nil {
-		log.Error().Err(err).Msg("failed to complete processing")
-		os.Exit(1)
+done:
+	for {
+		select {
+		case <-done:
+			log.Info().Msg("processing done")
+			break done
+		case <-cancelCtx.Done():
+			log.Info().Msg("processing cancelled")
+			break done
+		}
 	}
 
-	<-done
+	merr := <-merrs
+
+	return utilities.CombineErrors(err, merr)
```

Comment:
My understanding is when the context is cancelled, we do not wait for DoProcessFiles to complete, which means err will always be nil. DoProcessFiles needs to take and do something with the context as well, doesn't it?

Reply:
Changed to always wait for completion on cancel.